### PR TITLE
Enable hysteresis in water gas 2p model

### PR DESCRIPTION
**Pull request by totto82**
Needs more testing.

### CODE REVIEW

**Comment by totto82** [comment]
jenkins build this please

File: opm/material/fluidmatrixinteractions/EclHysteresisTwoPhaseLawParams.hpp [hunk 1]
```diff
@@ -194,6 +205,10 @@ class EclHysteresisTwoPhaseLawParams : public EnsureFinalized
                 Swcri_ = info.Sogcr;
                 Swmaxi_ = 1.0 - info.Sgl - info.Swl;
                 pcmaxi_ = info.maxPcgo;
+            } else if (twoPhaseSystem == EclTwoPhaseSystemType::GasWater) {
+                Swcri_ = info.Sgcr;
```
Comment (osae) [inline opm/material/fluidmatrixinteractions/EclHysteresisTwoPhaseLawParams.hpp:209]: Same as above.

**Review by osae — COMMENTED** [review]
Thank you for the contribution.  The GasWater tag makes it orthogonal to previous code, so the addition should be pretty safe regarding side-effects to existing functionality. Please consider the few comments, and I will be happy to approve.

File: opm/material/fluidmatrixinteractions/EclHysteresisTwoPhaseLawParams.hpp [hunk 2]
```diff
@@ -144,9 +149,12 @@ class EclHysteresisTwoPhaseLawParams : public EnsureFinalized
 
         // Additional Killough hysteresis model for pc
         if (config().pcHysteresisModel() == 0) {
-            if (twoPhaseSystem == EclTwoPhaseSystemType::GasOil) {
+            if (twoPhaseSystem == EclTwoPhaseSystemType::GasOil ) {
                 Swcrd_ = info.Sogcr;
                 pcmaxd_ = info.maxPcgo;
+            } else if (twoPhaseSystem == EclTwoPhaseSystemType::GasWater) {
+                Swcrd_ = info.Sgcr;
```
[the remark this answers is not in the document]
Comment (totto82) [reply opm/material/fluidmatrixinteractions/EclHysteresisTwoPhaseLawParams.hpp]: Thanks!

**Comment by osae** [comment]
Looks good to me. Approved.

**Pull request merged**